targets/TARGET_NORDIC/TARGET_NRF5/sleep.c

```diff
@@ -71,10 +77,20 @@ void hal_sleep(void)
             __WFE();
         }
     }
+
+    if (deepsleep) {
+        nrf_timer_task_trigger(NRF_TIMER1, NRF_TIMER_TASK_START);
```

Comment:
This should be conditional of the timer state before deepsleep entry. There is no reason to start the peripheral after deep sleep if it wasn't started earlier.

Reply:
If `MBED_TICKLESS` is enabled, then `usticker` is enabled (`RtosTimer`). It looks like all NRF5 boards have tick-less mode enabled, so `usticker` is always in use.

Reply:
I know I've commented about this issue earlier and I've made a mistake. `RtosTimer` use the lp_ticker, not the us_ticker: 

https://github.com/ARMmbed/mbed-os/blob/f1cf77fa443ee48b29861a8581f5d3f685c7ec2e/rtos/TARGET_CORTEX/mbed_rtx_idle.cpp#L50-L51

In this code `_ticker_data` is initialized by the `TimerEvent` constructor with the lp ticker. 

------

Considering defects or current implementation details in separate modules to build new code is a good way to build a fragile house of cards. It is expected  `hal_deepsleep` will restore us_ticker operating state at wakeup time so we should do just that. If the us ticker was started before deepsleep then start it again otherwise; do nothing.

Cleaning incorrect uses of the us ticker (event queue, `wait_ms`) will be a mandatory before this branch gets into master. Better to not add work we will have to do eventually.

Reply:
I this case I'll provide a fix to enable `usticker` only if it was enabled before going to sleep.

Reply:
Fixed.